src/components/common/dialogs/TaskDialog.js

```diff
@@ -0,0 +1,126 @@
+import React from 'react'
+import { Platform, View } from 'react-native'
+import { t } from '@lingui/macro'
+
+import { useTaskList } from '../../dashboard/Tasks/hooks/useTasks'
+import { Section, Text } from '../../common'
+import { withStyles } from '../../../lib/styles'
+
+const dialogStyles = ({ theme }) => {
+  return {
+    subTitleContainer: {
+      marginBottom: theme.sizes.defaultDouble,
+    },
+    subtitle: {
+      textAlign: 'center',
+      marginBottom: 25,
+    },
+    taskContainer: {
+      marginBottom: 30,
+      backgroundColor: theme.colors.secondaryGray,
+      borderRadius: 20,
+      marginLeft: 'auto',
+      marginRight: 'auto',
+      position: 'relative',
+      paddingTop: 40,
+
+      marginTop: 20,
+      ...Platform.select({
+        web: {
+          maxWidth: 'fit-content',
+          paddingTop: 60,
+          boxShadow: theme.shadows.shadow2,
+        },
+        native: {
+          shadowColor: 'rgba(0, 0, 0, 1)',
+          shadowOffset: { width: 8, height: 1 },
+          shadowRadius: 2.22,
+          shadowOpacity: 1,
+          elevation: 4,
+          paddingTop: 40,
+        },
+      }),
+    },
+    taskHeader: {
+      position: 'absolute',
+      display: 'flex',
+      justifyContent: 'center',
+      backgroundColor: theme.colors.green,
+
+      borderRadius: 50,
+      ...Platform.select({
+        web: {
+          top: -15,
+          left: 80,
+          height: 10,
+          padding: 16,
+          width: 150,
+        },
+        native: {
+          top: -20,
+          left: 75,
+          height: 30,
+          width: 120,
+          paddingLeft: 10,
+          paddingRight: 10,
+        },
+      }),
+    },
+    headerText: {
+      fontSize: 14,
+      ...Platform.select({
+        web: {
+          fontSize: 14,
+        },
+        native: {
+          fontSize: 12,
+        },
+      }),
+      color: 'white',
+      textTransform: 'uppercase',
+      fontFamily: theme.fonts.slab,
+    },
+    taskBody: {
+      flexDirection: 'column',
+    },
+    taskDesc: {
+      display: 'flex',
+      justifyContent: 'center',
+      marginBottom: 30,
+    },
+    taskAction: {
+      ...Platform.select({
+        web: {
+          width: '100%',
+        },
+      }),
+    },
+  }
+}
+
+export default withStyles(dialogStyles)(({ styles, theme }) => {
```

Comment:
remember we had issues with unnamed default exports ? let's use `const Component =` then `export default hoc()(Component)`